arlo-client/src/components/Audit/Setup/Participants/index.tsx

```diff
@@ -26,19 +31,45 @@ interface IProps {
 }
 
 const Participants: React.FC<IProps> = ({ audit, nextStage }: IProps) => {
+  const { electionId } = useParams()
   const [isLoading, setIsLoading] = useState(false)
+  const submit = async (values: IValues) => {
+    try {
+      setIsLoading(true)
+      /* istanbul ignore else */
+      if (values.csv) {
+        const formData: FormData = new FormData()
+        formData.append('jurisdictions', values.csv, values.csv.name)
+        const errorResponse: IErrorResponse = await api(
+          `/election/${electionId}/jurisdictions/file`,
+          {
+            method: 'PUT',
+            body: formData,
+          }
+        )
+        if (checkAndToast(errorResponse)) return
+      }
+      nextStage()
```

Comment:
I don't see where this is polling `…/jurisdictions/file` to ensure the upload is processed before moving on. Does it do that anywhere? Should it?

Reply:
@eventualbuddha It is not yet. When I got to that point, I realized that there is no mechanism in place to manage going from stage to stage in a controlled way. What if they click on the sidebar to jump ahead when the data isn't ready yet? What if they jump backwards and edit something and then things have to re-process before going forward? Polling whether a particular stage is ready to be accessed is part of that control flow, and I'm currently developing that in a separate branch, in order to keep PR size down.